### PR TITLE
feat(insights): compare legacy and hogql insights

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

We want to launch HogQL trend and lifecycle insights. We also want to make sure the data is the same between the two.

## Changes

Adds a management command that compares a set of lifecycle insight results. 

## How did you test this code?

I ran this in the python shell directly. I'll keep tweaking and running it like that for now. This PR is just a snapshot and likely won't get merged.

### CODE REVIEW

**Comment by mariusandra** [comment]
This is just a management command that's not run from anywhere, but is useful when comparing HogQL results between teams. Let's get it in, and possibly expand with other queries as we finish them.

File: posthog/management/commands/compare_hogql_insights.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    help = "Test if HogQL insights match their legacy counterparts"
+
+    def handle(self, *args, **options):
+        from posthog.models import Insight, Filter
+        from posthog.queries.trends.trends import Trends
+        from posthog.hogql_queries.legacy_compatibility.filter_to_query import filter_to_query
+        from posthog.hogql_queries.query_runner import get_query_runner
+
+        insights = (
+            Insight.objects.filter(filters__contains={"insight": "LIFECYCLE"}, saved=True, deleted=False, team_id=2)
```
Comment (webjunkie) [inline posthog/management/commands/compare_hogql_insights.py:16]: Why team 2? For me it's 1.

**Review by webjunkie — COMMENTED** [review]
I made a lifecycle insight and got
`posthog.hogql.errors.HogQLException: Unknown CompareOperationOp: InCohort`
Not sure what that is.

But, yeah, can still be a useful command.

File: posthog/management/commands/compare_hogql_insights.py (hunk 1 above)
Comment (mariusandra) [reply posthog/management/commands/compare_hogql_insights.py:16]: That's what we have in production on the US cloud. The next version of this query should just check for all teams.

**Comment by mariusandra** [comment]
> I made a lifecycle insight and got `posthog.hogql.errors.HogQLException: Unknown CompareOperationOp: InCohort` Not sure what that is.

Oh, seems like you found a bug: lifecycle insights with cohort filters fail. Will fix.

**Comment by webjunkie** [comment]
> > I made a lifecycle insight and got `posthog.hogql.errors.HogQLException: Unknown CompareOperationOp: InCohort` Not sure what that is.
> 
> Oh, seems like you found a bug: lifecycle insights with cohort filters fail. Will fix.

Ah, I just got super confused since it works in the frontend. Turns out it's one of my insights, but not the one I created right before.

**Comment by mariusandra** [comment]
Fixed here: https://github.com/PostHog/posthog/pull/19501

**Review by webjunkie — APPROVED** [review]
Runs now for me. Let's get it in.

**Pull request merged**